components/brave_wallet/browser/keyring_service.cc

```diff
@@ -550,7 +551,9 @@ KeyringService::KeyringService(JsonRpcService* json_rpc_service,
       profile_prefs_(profile_prefs),
       local_state_(local_state) {
   DCHECK(profile_prefs);
-  DCHECK(local_state);
+  if (!local_state) {
```

Comment:
KeyringService is created during unit tests?

Reply:
oh, apparently this was a recent change. Maybe we can kill two birds with one stone and just not create the BraveWalletService if local state is null?

Reply:
That fixes several issues here I think including BraveWalletP3A above

Reply:
you could implement `ServiceIsNULLWhileTesting` to return true if `g_browser_process->local_state()` is nullptr and CHECK_IS_TEST there

Reply:
Done in 23753c0c89ca5fa69dc240947044c05a600355ff

Reply:
is this check still needed or can it revert back to DCHECK?

Reply:
Reverted in [48842cf](https://github.com/brave/brave-core/pull/20433/commits/48842cfd49ed07c67bb75a4f1e57f88f22961eb7), let's see if anything fails.

Reply:
No problems on CI after reverting this.